scripts/users.js

```diff
@@ -0,0 +1,275 @@
+const addressRegex = /\w{1,300}/;
+
+const getDatabase = () => {
+  return JSON.parse(localStorage.getItem("database"));
+};
+
+const getUsers = () => {
+  return JSON.parse(localStorage.getItem("database")).users;
+};
+
+const sanitize = (string) => {
+  return string
+    .replace(/&/g, "&amp;")
+    .replace(/</g, "&lt;")
+    .replace(/"/g, "&quot;");
+};
+
+const addUserToTable = (user) => {
+  let tableRow = document.createElement("tr");
+
+  let idTd = document.createElement("td");
+  let nameTd = document.createElement("td");
+  let emailTd = document.createElement("td");
+  let ageTd = document.createElement("td");
+  let addressTd = document.createElement("td");
+  let editTd = document.createElement("td");
+  let deleteTd = document.createElement("td");
+
+  let editButton = document.createElement("button");
+  let deleteButton = document.createElement("button");
+
+  editButton.innerHTML = "Edit";
+  editButton.className = "button";
+  editButton.id = "editButton";
+
+  deleteButton.innerHTML = "Delete";
+  deleteButton.className = "button";
+  deleteButton.id = "deleteButton";
+
+  editTd.appendChild(editButton);
+  deleteTd.appendChild(deleteButton);
+
+  idTd.innerHTML = sanitize(user.id);
+  nameTd.innerHTML = sanitize(user.name);
+  emailTd.innerHTML = sanitize(user.email);
+  ageTd.innerHTML = sanitize(user.age.toString());
+  addressTd.innerHTML = sanitize(user.address);
+
+  tableRow.appendChild(idTd);
+  tableRow.appendChild(nameTd);
+  tableRow.appendChild(emailTd);
+  tableRow.appendChild(ageTd);
+  tableRow.appendChild(addressTd);
+  tableRow.appendChild(editTd);
+  tableRow.appendChild(deleteTd);
+
+  tableRow.id = user.id;
+
+  tableBody.appendChild(tableRow);
+};
+
+getUsers().forEach((usersObj) => {
+  addUserToTable(usersObj);
+});
+
+document.getElementById("usersTable").onclick = (event) => {
+  if (event.target.id !== "editButton" && event.target.id !== "deleteButton") {
+    return;
+  }
+
+  let userId = event.target.parentElement.parentElement.children[0].innerHTML;
+  let user = getUsers().find((u) => u.id === userId);
+
+  if (event.target.id == "editButton") {
+    loadUserToForm(user);
+  } else if (event.target.id == "deleteButton") {
+    console.log(1);
+    deleteUser(user);
+  }
+};
+
+const addUserToJSON = (user) => {
+  let users = getUsers();
+  users.push(user);
+  localStorage.setItem(
+    "database",
+    JSON.stringify({
+      ...getDatabase(),
+      users,
+    })
+  );
+};
+
+form.addEventListener("submit", (ev) => {
+  ev.preventDefault();
+
+  let id = ev.target.id.value;
+
+  let newUserId = (Math.random() * 10)
+    .toString()
+    .replace(".", "")
+    .substr(0, 10);
+  let name = ev.target.name.value;
+  let email = ev.target.email.value;
+  let age = ev.target.age.value;
+  let address = ev.target.address.value;
+
```

Comment:
You have some duplications below and it's a good idea to have them minimizied. E.g. the age check is done twice and you can easily extract that particular check in a function with appropriate name then reuse it. The other thing is setting the errors. You could have functions for setting errors for particular field like
```javascript
const setErrorMessage = (target, error) => {
  target.innerHTML = error;
}
const setNameError = (error) => {
  setErrorMessage(nameError, error);
}
etc...
```
then just call it
```javascript
    if (!nameRegex.test(name)) {
      setNameError('error message...');
    } else {
      setNameError('');
    }
```
The other thing is that these checks can be encapsulated in functions and this way the code here would become more clear and easy to read and if I want to know the internals, only then I can go check particular validation function.
```javascript
const validateName = () => {
  if (!nameRegex.test(name)) {
      setNameError('error message...');
    } else {
      setNameError('');
    }
}
```
then you will have 
```javascript
validateName();
validateEmail();
...
```
And further you can push these validator functions in array and call them sequentially and this way decouple the validation.
```javascript
const validators = [];
const registerValidator = (validator) => {
  validators.push(validator);
}
const validateUserDetails = () => {
  validators.forEach(validator => validator && validator());
}
// register validators
registerValidator(validateName);
registerValidator(validateEmail);
// then validate the form
validateUserDetails();
```